apiserver/root.go

```diff
@@ -488,6 +488,23 @@ func (ctx *facadeContext) LeadershipPinner(modelUUID string) (leadership.Pinner,
 	return leadershipPinner{pinner}, nil
 }
 
+// LeadershipReader is part of the facade.Context interface.
+// It returns a reader that can be used to return all application leaders
+// in the model.
+func (ctx *facadeContext) LeadershipReader(modelUUID string) (leadership.Reader, error) {
```

Comment:
why is this on root.go instead of something that relates to Leadership ?

Reply:
`facadeContext` has the lease manager, from which the reader is returned, same as for claimer and pinner.